src/elements/content-sidebar/activity-feed/common/user-link/UserLink.tsx

```diff
@@ -0,0 +1,56 @@
+import * as React from 'react';
+import classnames from 'classnames';
+import { Link } from '../../../../../components/link';
+import { GetProfileUrlCallback } from '../../../../common/flowTypes';
+import './UserLink.scss';
+
+export interface UserLinkProps {
+    className?: string;
+    getUserProfileUrl?: GetProfileUrlCallback;
+    id: string;
+    name: string;
+}
+
+interface State {
+    profileUrl?: string;
+}
+
+class UserLink extends React.PureComponent<UserLinkProps, State> {
+    state: State = {
+        profileUrl: undefined,
+    };
+
+    getProfileUrlHandler = (profileUrl?: string) => {
+        this.setState({
+            profileUrl,
+        });
+    };
+
+    getUserProfileUrl() {
+        const { id, getUserProfileUrl } = this.props;
+        if (!getUserProfileUrl) {
+            return Promise.resolve();
+        }
+
+        return getUserProfileUrl(id).then(this.getProfileUrlHandler);
+    }
+
+    componentDidMount() {
+        this.getUserProfileUrl();
+    }
+
+    render() {
+        const { name, getUserProfileUrl, className, ...rest } = this.props; // eslint-disable-line @typescript-eslint/no-unused-vars
```

Comment:
Add a comment that this is to prevent `getUserProfileUrl` from being passed to down to the `Link` component